todo_app/data/trello_client.py

```diff
@@ -0,0 +1,92 @@
+import os
+import requests
+
+TRELLO_KEY = os.environ.get('TRELLO_API_KEY', '')
+TRELLO_TOKEN = os.environ.get('TRELLO_API_TOKEN', '')
+TRELLO_TO_DO_BOARD_ID = os.environ.get('TRELLO_TO_DO_BOARD_ID', '')
+
+LISTS_URL = f"https://api.trello.com/1/boards/{TRELLO_TO_DO_BOARD_ID}/lists"
+CARDS_URL = f"https://api.trello.com/1/cards"
+card_request_params = {
+    'key': TRELLO_KEY,
+    'token': TRELLO_TOKEN,
+    'cards': 'open',
+    'card_fields': 'id,name,desc,due',
+}
+list_request_params = {
+    'key': TRELLO_KEY,
+    'token': TRELLO_TOKEN,
+}
+
+
+class TrelloClient:
+
+    @classmethod
+    def get_trello_lists(cls):
+        """
+        Fetches the list of all lists available in the Trello
+        Returns:
+            The json response from the Trello API (the list of Trello lists if successful)
+        """
+        return requests.get(LISTS_URL, params=card_request_params).json()
+
+    @classmethod
+    def _get_list_id(cls, list_name):
+        """
+        Utility function to get the id of the given list
+        Args:
+            list_name: the name of the list
+
+        Returns:
+            The id of the list
+        """
+        return [card_list['id'] for card_list in cls.get_trello_lists() if card_list['name'] == list_name][0]
```

Comment:
This works nicely, but worth considering whether the flexibility it gives us is actually worth something or whether a simpler "hardcoding" would work more nicely, e.g. asking the user to provide a `TRELLO_TO_DO_LIST_ID` environment variable, and just using that if they want the To Do/Not Started list

Yours handles some level of set up for us, but does also add an extra network request in to each call to this function

Reply:
Funnily enough I was actually planning on going in the opposite direction and adding a look-up for the board ID as well but forgot!

I'd added in the list id lookup to make it easy for us to create extra statuses etc. It feels odd to have to expose to the user the implementation detail that each status is implemented as a list when we could have used e.g. Trello Labels or custom fields (and might do so in the future). But it's a good point about the extra network call each time - I'll refactor to get those once and cache them.